### DISCUSSION TITLE
`total_time` calculation

### DISCUSSION THREAD

**Discussion by aodj** [IBM/ada#7] [Q&A]
Hi, I'm very interested in Ada, and quite like the idea behind it. I'm curious if you could elaborate on some of the design decisions, specifically the user of `EXTRACT(EPOCH FROM (end_date - start_date))/60 AS total_time` in the queries. Looking at the `task_instance` table, it has a `duration` column with the run time in seconds.

Here's an example of what I mean:

```
+------------------+-------------+-----------------------------------------------------+
|total_time        |duration     |total_time_interval                                  |
+------------------+-------------+-----------------------------------------------------+
|8.062844333333334 |483.77066    |0 years 0 mons 0 days 0 hours 8 mins 3.77066 secs    |
|13.423839833333334|805.43039    |0 years 0 mons 0 days 0 hours 13 mins 25.43039 secs  |
|1.7442844000000002|104.657064   |0 years 0 mons 0 days 0 hours 1 mins 44.657064 secs  |
|1.6915136166666669|101.490817   |0 years 0 mons 0 days 0 hours 1 mins 41.490817 secs  |
|8.685444916666667 |521.126695   |0 years 0 mons 0 days 0 hours 8 mins 41.126695 secs  |
|1.5417908999999999|92.507454    |0 years 0 mons 0 days 0 hours 1 mins 32.507454 secs  |
+------------------+-------------+-----------------------------------------------------+
```

**Comment by juliaalfarias** [comment]
Hey @aodj!

Thanks a bunch for checking out ADA and taking the time to suggest improvements! Your feedback is really valuable to me. You’re absolutely right – I overcomplicated things with that unnecessary calculation.

At first, I added that calculation to make things easier to understand, so people wouldn’t need to dig deep into the data details. But, like you pointed out, this approach led to redundancy. Your suggestion has a lot of potential to optimize ADA, so I’d like to invite you to contribute with us if you’re interested. Your input would be highly appreciated and quite exciting for us!

Thanks again! 😄
